### ISSUE TITLE
Entity shakes if standing on entity

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#6076]
If a player or any other collidable entity stands on top of another collidable entity, it starts to visible shake vertically. This is very weird and seems to break collision physics somewhat.

This can be easily tested by using Mobs Redo with any mob. Just stand on top of a mob and look on your screen. It visibly starts shaking vertically. As if the player can't decide whether to fall or not.

You can also do this with two mobs if you manage to stack them.

Tested in 0.4.16.

**Comment by bigfoot547** [comment]
I noticed this happen before the sneeking changes

**Comment by Wuzzy2** [comment]
Yeah, this bug is in Minetest for a very long time, I just did not bother to report it earlier.

**Comment by HybridDog** [comment]
l think it's a bug of the collision algorithm. It should be reworked that to look like in terasology https://www.youtube.com/watch?v=1pWBrGHS8H8&t=408.

**Comment by azekillDIABLO** [comment]
It's a bug i've ALWAYS seen in minetest. I never really thought it was a bug, i think it comes from the fact that your position is actualized every ~0.02 seconds and that cause a shake when you're on the entity.

**Comment by paramat** [comment]
I don't consider this a bug, entities are subject to gravity so try to fall but are stopped by collision boxes.
Item entites actually shake too if you remove the special code that gives them velocity and acceleration zero when they are stationary.
Entity standing on entity seems not a very common situation so this doesn't seem a big issue. The shaking can be quite charming in the way it suggests 'tension' between 2 touching entities.

**Comment by Wuzzy2** [comment]
> The shaking can be quite charming in the way it suggests 'tension' between 2 touching entities.

No, this is just a lame excuse. It is not charming at all, it looks horrible in my eyes and looks like a glitch (because it *is* a gitch). There is no logical reason why entities should do something like this at all.

Yes, it's not very common and probably not very important, but still a bug.

**Comment by paramat** [comment]
No it's not an excuse, it's my opinion, however i admit i don't know how extreme the vibration is, what is the vertical amplitude?
Entity on entity is rare.

**Comment by Wuzzy2** [comment]
I don't know exactly, but it's enough to be annoying for me. Just try it out for yourselves, it's not hard to reproduce this.

**Comment by paramat** [comment]
How many stadard texture pixels? I would have to downoad a mob mod to see. I tried with dropped items and they pass through each other.

**Comment by DS-Minetest** [comment]
I've tested with an entity:
```lua
minetest.register_entity("test_obj_shake:ent", {
	physical = true,
	collide_with_objects = true,
	collisionbox = {-0.5, -0.5, -0.5, 0.5, 0.5, 0.5},
	visual = "cube",
	on_activate = function(self, staticdata, dtime_s)
		self.object:set_acceleration(vector.new(9.81, 0, 0))
	end,
})
```
It has an acceleration of the default value of the `"movement_gravity"` setting (`9.81`).
![screenshot_20170825_213505](https://user-images.githubusercontent.com/7613443/29729842-1e8d4f4a-89de-11e7-9e3a-3cab43bf5c7d.png)

Actually with smaller values, eg. `1`, it vibrates slower but further:
![screenshot_20170825_213239](https://user-images.githubusercontent.com/7613443/29729914-658a6a4a-89de-11e7-98d5-9f8203385f3f.png)

With `0.5` the both objects _melt_ together into the same pos and it looks as if there was only one.

**Comment by paramat** [comment]
Will test.

**Comment by Megaf** [comment]
Wont fix :)
The whole entity thing and mob thing must be rewritten IMHO.

**Comment by paramat** [comment]
? We'll fix it if we can.
Big rewrites are usually a bad idea.
I doubt you have enough code knowledge to judge a rewrite is needed.
Your comment is baseless negativity.

**Comment by Megaf** [comment]
Hah, sorry.
Let me elaborate then.
The issue lies deep in what we call entities, everything that is not player, particle or static object are entities. That include text in signs, UFOs, bots, dropped item and mobs.

I do believe that these things should have their one space and code and be handled independently by the server.

For instance, a /clearobjects will remove all UFOs, boats, cars and sheeps a player has, will wipe (temporarily) all signs.

In order to fix this, mobs should be a thing on their own, they should be permanent and work in the same way they do in that other voxel game.

The shake described in this issue I think it happens in the same frequency of the server step, so it has to do with the mob getting updated, similar shake happens if you stand on a piston and activates it. Though then it will be just one "shake".

Again, things must be rewritten.
@paramat And you are absolutely right. I have no idea about coding. I'm just being purely logical.

**Comment by paramat** [comment]
No problem, i was a bit grumpy.

**Comment by GreenXenith** [comment]
This seems to be fixed in 5.0.

**Comment by DS-Minetest** [comment]
>This seems to be fixed in 5.0.

I can confirm this for version 6ed9c6f.

**Comment by SmallJoker** [comment]
Fixed since https://github.com/minetest/minetest/commit/ee63b94f2c#diff-746d6155906a987734997d4cad358153R383
I didn't notice there was an issue about this problem already. :smile:
